cpp/src/arrow/flight/sql/odbc/flight_sql/CMakeLists.txt

```
@@ -0,0 +1,209 @@
+
+if(MSVC)
+  # the following definitions stop arrow from using __declspec when staticly
+  # linking and will break on Windows without them
+  add_compile_definitions(ARROW_STATIC ARROW_FLIGHT_STATIC)
+endif()
+
+enable_testing()
+
+add_library(arrow_odbc_spi_impl
+            include/flight_sql/flight_sql_driver.h
+            accessors/binary_array_accessor.cc
+            accessors/binary_array_accessor.h
+            accessors/boolean_array_accessor.cc
+            accessors/boolean_array_accessor.h
+            accessors/common.h
+            accessors/date_array_accessor.cc
+            accessors/date_array_accessor.h
+            accessors/decimal_array_accessor.cc
+            accessors/decimal_array_accessor.h
+            accessors/main.h
+            accessors/primitive_array_accessor.cc
+            accessors/primitive_array_accessor.h
+            accessors/string_array_accessor.cc
+            accessors/string_array_accessor.h
+            accessors/time_array_accessor.cc
+            accessors/time_array_accessor.h
+            accessors/timestamp_array_accessor.cc
+            accessors/timestamp_array_accessor.h
+            address_info.cc
+            address_info.h
+            flight_sql_auth_method.cc
+            flight_sql_auth_method.h
+            flight_sql_connection.cc
+            flight_sql_connection.h
+            flight_sql_driver.cc
+            flight_sql_get_tables_reader.cc
+            flight_sql_get_tables_reader.h
+            flight_sql_get_type_info_reader.cc
+            flight_sql_get_type_info_reader.h
+            flight_sql_result_set.cc
+            flight_sql_result_set.h
+            flight_sql_result_set_accessors.cc
+            flight_sql_result_set_accessors.h
+            flight_sql_result_set_column.cc
+            flight_sql_result_set_column.h
+            flight_sql_result_set_metadata.cc
+            flight_sql_result_set_metadata.h
+            flight_sql_ssl_config.cc
+            flight_sql_ssl_config.h
+            flight_sql_statement.cc
+            flight_sql_statement.h
+            flight_sql_statement_get_columns.cc
+            flight_sql_statement_get_columns.h
+            flight_sql_statement_get_tables.cc
+            flight_sql_statement_get_tables.h
+            flight_sql_statement_get_type_info.cc
+            flight_sql_statement_get_type_info.h
+            flight_sql_stream_chunk_buffer.cc
+            flight_sql_stream_chunk_buffer.h
+            get_info_cache.cc
+            get_info_cache.h
+            json_converter.cc
+            json_converter.h
+            record_batch_transformer.cc
+            record_batch_transformer.h
+            scalar_function_reporter.cc
+            scalar_function_reporter.h
+            system_trust_store.cc
+            system_trust_store.h
+            utils.cc)
+target_include_directories(arrow_odbc_spi_impl PUBLIC ${CMAKE_CURRENT_LIST_DIR})
+if(WIN32)
+  target_sources(arrow_odbc_spi_impl
```

```suggestion
  target_sources(arrow_odbc_spi_impl PRIVATE
```
According to cmake docs, [`target_sources`](https://cmake.org/cmake/help/latest/command/target_sources.html) need a `INTERFACE`, `PUBLIC` or `PRIVATE` keyword to be able to build. I feel `PRIVATE` might be suitable here

---

Adding the keyword resolved some build issues from my local environment